File: Backend/Spread.Api/Domain/Companies/CompanyProfile.cs
using Spread.Api.Domain.Assets;

namespace Spread.Api.Domain.Companies;

public sealed record CompanyProfile(
    string Ticker,
    string CompanyName,
    AssetType AssetType,
    string? Sector,
    string? Industry,
    string? Exchange,
    string? Currency,
    string? Country,
    decimal? MarketCapitalization,
    decimal? Beta,
    bool IsActivelyTrading,
    string? Website,
    string? LogoUrl,
    DateTimeOffset FetchedAt,
    string Provider);

File: Frontend/lib/company-profile.ts
export type CompanyProfile = {
  ticker: string;
  companyName: string;
  assetType: string;
  sector: string | null;
  industry: string | null;
  exchange: string | null;
  currency: string | null;
  country: string | null;
  marketCapitalization: number | null;
  beta: number | null;
  isActivelyTrading: boolean;
  website: string | null;
  logoUrl: string | null;
  fetchedAt: string;
  provider: string;
};
